Index: public/assets/javascripts/ckeditor/plugins/studip-upload/plugin.js
===================================================================
--- public/assets/javascripts/ckeditor/plugins/studip-upload/plugin.js	(revision 50928)
+++ public/assets/javascripts/ckeditor/plugins/studip-upload/plugin.js	(working copy)
@@ -54,11 +54,10 @@
                     alert(lang.uploadError + '\n\n' + errors.join('\n'));
                 }
             };
-
         // actual file upload handler
         // NOTE depends on jQuery File Upload plugin being loaded beforehand!
         // TODO integrate jQuery File Upload plugin into studip-upload
-        var inputId = 'fileupload';
+        
         editor.on('instanceReady', function(event){
             var $container = $(event.editor.container.$);
 
@@ -65,7 +64,7 @@
             // install upload handler
             $('<input>')
                 .attr({
-                    id: inputId,
+                    class: "fileupload",
                     type: 'file',
                     name: 'files[]',
                     multiple: true
@@ -95,9 +94,8 @@
 
         // avoid multiple uploads of the same file via drag and drop
         editor.on('beforeDestroy', function(event){
-            if ($.fn.fileupload) {
-                $('#' + inputId).fileupload('destroy');
-            }
+            $(event.editor.container.$).find('.fileupload').fileupload('destroy');
+
         });
 
         // ckeditor
@@ -105,7 +103,7 @@
             exec: function(editor){
                 // NOTE if  $('#' + inputId) is stored in variable then
                 //      upload works only once
-                $('#' + inputId).click();
+                $(editor.container.$).find('.fileupload').click();
             }
         });
         editor.ui.addButton('upload', {  // toolbar button
